### PR TITLE
Ask for physics update when placing flowing liquids from biome decorator...

### PR DESCRIPTION
**Pull request by jimmikaelkael**
Fix attempt for #27.
Currently untested.

### CODE REVIEW

**Comment by jimmikaelkael** [comment]
Just tested it, this fix doesn't work...
@deathcap @mastercoms  any idea ?

**Comment by mastercoms** [comment]
receivePulse needs to be called.

**Comment by jimmikaelkael** [comment]
Shouldn't this be done automatically when calling org.bukkit.block.BlockState update method and asking for physics update ?

**Comment by mastercoms** [comment]
Yeah, we should modify it in glowkit, but I'll pull request a quick fix for now.

**Comment by mastercoms** [comment]
Closing this as a better fix can be done in Glowkit.

File: src/main/java/net/glowstone/block/GlowBlock.java [hunk 1]
```diff
@@ -452,5 +461,6 @@ public void applyPhysics(Material oldType, int newTypeId, byte oldData, byte new
         if (type != null) {
             type.onBlockChanged(this, oldType, oldData, newType, newData);
         }
+        type.receivePulse(this);
```
Comment (jimmikaelkael) [inline src/main/java/net/glowstone/block/GlowBlock.java]: Not what I expected. But it should work. Any reason we pulse even if applyPhysics is set to false ?

**Comment by mastercoms** [comment]
I don't know the test case for this, so could you test for me and tell the result?

**Comment by jimmikaelkael** [comment]
Not currently I'm not at home :)
On my side I was thinking about inserting the pulse in GlowBlock, after line 455 (current HEAD), and maybe line 462.

I think that's how it worked in bukkit but I'm really not sure.

File: src/main/java/net/glowstone/block/GlowBlock.java (hunk 1 above)
Comment (mastercoms) [inline src/main/java/net/glowstone/block/GlowBlock.java]: Well `.update()` with no applyPhysics still tries to update the block, so we should still pulse the block when it is updated.

**Comment by jimmikaelkael** [comment]
Well I'll try the current fix, let me a few minutes to install all what's needed.

**Comment by mastercoms** [comment]
Ok, I pushed your suggestion. When you're ready, merge it into your branch.

**Comment by mastercoms** [comment]
I made an oopsie. Instead of using notifyType, I used type, which didn't exist. I pushed a new thing, pull request it. Sorry.

**Comment by jimmikaelkael** [comment]
Yeah :) It can happen, no problem.

File: src/main/java/net/glowstone/block/GlowBlock.java [hunk 2]
```diff
@@ -451,6 +461,8 @@ public void applyPhysics(Material oldType, int newTypeId, byte oldData, byte new
         BlockType type = itemTable.getBlock(oldType);
         if (type != null) {
             type.onBlockChanged(this, oldType, oldData, newType, newData);
+            type.receivePulse(this);
         }
+
```
Comment (mastercoms) [inline src/main/java/net/glowstone/block/GlowBlock.java]: also accidental new line

**Comment by jimmikaelkael** [comment]
I'm testing, if it works I'll remove the newline ;)

**Comment by jimmikaelkael** [comment]
Had an exception loop triggering stack overflow, no way to see what's the error :)
I'm reverting to your fix.

**Comment by jimmikaelkael** [comment]
@mastercoms  Well, your initial fix is what works, the other method I suggested either doesn't work, either throw a loop of exceptions.

The current results:
flowing liquids

I guessed why they make lava flowing farther and faster in the nether: I think it's to be able to form the lava falls more quickly to the ground once a chunk is populated, so that the player see it well formed.

I fixed one of my oddities in the latest commit, that was when playing with structures. The only thing I haven't be able to check is the structures that fit on several chunks (there is currently only one: the desert temple). But anyway, that will be the structures that will need a fix if it's broken.

**Comment by jimmikaelkael** [comment]
I've rebased and  cleaned up the PR.

**Pull request merged**